articles/implementing-aws-s3-functionalities-on-a-nodejs-graphql-api/index.md

```diff
@@ -0,0 +1,624 @@
+
+- [Uploading multiple objects to s3 bucket](#uploading-multiple-objects-to-s3-bucket)
+
+- [Fetching uploaded objects from s3 bucket](#fetching-uploaded-objects-from-s3-bucket)
+
+- [Deleting a single object from s3 bucket](#deleting-a-single-object-from-s3-bucket)
+
+- [Deleting multiple objects from s3 bucket](#deleting-multiple-objects-from-s3-bucket)
+
+- [Deleting a s3 bucket](#deleting-a-s3-bucket)
+
+### Obtaining your security credentials.
+
+To obtain your security credentials, follow the following steps:
+
+- Head over to [AWS console](https://aws.amazon.com/console/).
+
+- On the top right corner, click `Sign in to the console`.
+
+- Enter your email address and click `Next`.
+
+- Enter your password and click `Sign in`.
+
+- In the redirected page, search for s3 in the search bar and click the first result.
+
+- You will be redirected to your s3 account page.
+
+- On the top right, there is your username and a dropdown arrow. Click here and then click `My security credentials`.
+
+- On the new page, click on `Access keys (access key ID and secret access key)`.
+
+- If you already have an `Access Key ID` and a `Secret access key` feel free to use them. But if you don't have one, click the `Create New Access Key` button.
+
+- Download the `Key File`.
+
+- From the current URL, you pick your region. For example, considering this URL: `https://console.aws.amazon.com/iam/home?region=us-east-2#/security_credentials`, the region is set to us-east-2.
+
+- In the `Key File` you downloaded, add the following:
+
+```javascript
+region=//the region from the URL. e.g us-east-2
+```
+
+- After setting the region, you are good for the next step.
+
+### Setting up the development server
+
+To set up the development server, clone this [Github repository](https://github.com/mwangiKibui/aws-sdk-node.js-s3-graphql). In the repository, in the `src` directory, there is the start and final folder. Throughout the article, we will be working on the start folder. In case you encounter an error or you want to compare your code, feel free to check the final folder.
+
+From the `Key File` you downloaded copy the data to the `.env` file appropriately. Ensure you match the names so that there is no conflict.
+
+In the cloned folder, run the following command to install the necessary dependencies:
+
+```bash
+npm install
+```
+
+Proceed to the start folder and go through the schema files to understand how we are structuring the data. Our focus throughout the article will be on the resolvers folder. It is therefore helpful to have a grasp of the underlying schema.
+
+### Creating an s3 bucket
+
+An s3 bucket is the foundational storage instance in s3. It consists of folders and objects. The objects are the files. Data in the bucket is stored as either a folder or an object.
+
+To create a bucket, we implement the following functionality in the `resolvers/mutation-resolvers.js` under `createBucket()` function:
+
+```javascript
+//create a bucket.
```

Comment:
It would be recommended to add useful code comments in this code snippet

Reply:
fixed.